tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -533,6 +533,23 @@ Status StorageManager::array_consolidate(
     config = &config_;
   }
 
+  // Get encryption key from config
+  std::string encryption_key_from_cfg;
+  std::string encryption_type_from_cfg;
+  if (!encryption_key) {
+    bool found = false;
+    encryption_key_from_cfg = config->get("sm.encryption_key", &found);
+    assert(found);
+    found = false;
+    encryption_type_from_cfg = config->get("sm.encryption_type", &found);
+    assert(found);
+    encryption_type_enum(encryption_type_from_cfg, &encryption_type);
```

Comment:
Doing this work is premature if `encryption_key_from_cfg` is empty. Only do this after we know that `encryption_key_from_cfg` is non-empty.